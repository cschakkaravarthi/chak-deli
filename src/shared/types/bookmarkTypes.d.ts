import Bookmark from '../models/Bookmark.model';
export declare const LOAD_BOOKMARKS = "LOAD_BOOKMARKS";
export declare const FETCH_BOOKMARKS = "FETCH_BOOKMARKS";
export declare const SET_BOOKMARK = "SET_BOOKMARK";
export declare const SAVE_BOOKMARK = "SAVE_BOOKMARK";
export declare const DELETE_BOOKMARK = "DELETE_BOOKMARK";
export declare const REMOVE_BOOKMARK = "REMOVE_BOOKMARK";
export interface LoadBookmarksAction {
    type: typeof LOAD_BOOKMARKS;
    bookmarks: Bookmark[];
}
export interface SetBookmarkAction {
    type: typeof SET_BOOKMARK;
    success: boolean;
}
export interface DeleteBookmarkAction {
    type: typeof DELETE_BOOKMARK;
    success: boolean;
}
export interface BookmarkState {
    bookmarks: Bookmark[];
    success: boolean;
}
//# sourceMappingURL=bookmarkTypes.d.ts.map