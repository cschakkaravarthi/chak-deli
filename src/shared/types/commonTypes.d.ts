export declare const TOGGLE_GLOBAL_TOAST = "TOGGLE_GLOBAL_TOAST";
export interface ToggleGlobalToastAction {
    type: typeof TOGGLE_GLOBAL_TOAST;
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
export interface CommonState {
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
//# sourceMappingURL=commonTypes.d.ts.map