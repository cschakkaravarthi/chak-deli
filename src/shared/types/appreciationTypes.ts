import { AppreciationCardNotificationModel as AlertModel } from '../models/Notification.model';
import { AppreciationCardTemplate, AppreciationCardsListGrouped } from '../models/AppreciationCards.model';

export const FETCH_APPRECIATION_DATA = 'FETCH_APPRECIATION_DATA';
export const LOAD_APPRECIATION_ALERT = 'LOAD_APPRECIATION_ALERT';
export const LOAD_APPRECIATION_ALERT_ERROR = 'LOAD_APPRECIATION_ALERT_ERROR';
export const CLEAR_APPRECIATION_REDUCER = 'CLEAR_APPRECIATION_REDUCER';
export const LOAD_APPRECIATION_MANAGER_ALERT = 'LOAD_APPRECIATION_MANAGER_ALERT';
export const SET_NEW_CARD_DETAIL = 'SET_NEW_CARD_DETAIL';
export const FETCH_CARD_TEMPLATES = 'FETCH_CARD_TEMPLATES';
export const CLEAR_NEW_CARD_FORM = 'CLEAR_NEW_CARD_FORM';
export const TOGGLE_SUCCESS_SENT_ALERT = 'TOGGLE_SUCCESS_SENT_ALERT';
export const GET_APPRECIATION_CARDS_LIST_GROUPED = 'GET_APPRECIATION_CARDS_LIST_GROUPED';
export const VIEW_APPRECIATION_CARD = 'VIEW_APPRECIATION_CARD';
export const APPRECIATION_CARD_ID = 'APPRECIATION_CARD_ID';
export const ARCHIVE_ALERT_STATE = 'ARCHIVE_ALERT_STATE';
export const LOAD_CARDS_IN_GROUP = 'LOAD_CARDS_IN_GROUP';

export interface AppreciationState {
  cardId: string;
  cardArchiveSuccess: boolean;
  isFetching: boolean;
  appreciationCardAlert?: AlertModel[];
  appreciationCardAlertError?: boolean;
  appreciationManagerCardAlert?: AlertModel[];
  newAppreciationCardForm: NewAppreciationCardForm;
  cardTemplates: AppreciationCardTemplate[];
  showSentSucessAlert: boolean;
  appreciationCardsListGrouped?: AppreciationCardsListGrouped;
  cardToView?: AlertModel;
  displayCardView: boolean;
}

export interface NewAppreciationCardForm {
  senderName?: string;
  templateId?: string;
  cardMessage?: string;
  recipientEmail?: string;
  recipientLastName?: string;
  recipientFirstName?: string;
  recipientSupervisorEmail?: string;
  managerName?: string;
}

export interface ArchiveAlertState {
  cardArchiveSuccess: boolean;
  type: typeof ARCHIVE_ALERT_STATE;
}

export interface SetArchiveAlert {
  cardId: string;
  type: typeof APPRECIATION_CARD_ID;
}

export interface FetchRequest {
  isFetching: boolean;
  type: typeof FETCH_APPRECIATION_DATA;
}

export interface LoadAppreciationAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT;
  appreciationCardAlert: AlertModel[];
}

export interface LoadAppreciationManagerAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_MANAGER_ALERT;
  appreciationManagerCardAlert: AlertModel[];
}

export interface AppreciationCardAlertError {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT_ERROR;
  appreciationCardAlertError?: boolean;
}

export interface ClearAprreciationAction {
  type: typeof CLEAR_APPRECIATION_REDUCER;
}

export interface SetNewCardDetailAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface SetDetailGroupAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface FetchAppreciationCardTemplatesAction {
  type: typeof FETCH_CARD_TEMPLATES;
  cardTemplates: AppreciationCardTemplate[];
}

export interface ClearNewCardFormAction {
  type: typeof CLEAR_NEW_CARD_FORM;
}

export interface ToggleSentSuccessAlert {
  type: typeof TOGGLE_SUCCESS_SENT_ALERT;
  showSentSucessAlert: boolean;
}

export interface GetAppreciationCardsListGroupedAction {
  type: typeof GET_APPRECIATION_CARDS_LIST_GROUPED;
  appreciationCardsListGrouped: AppreciationCardsListGrouped;
}

export interface ViewAppreciationCardAction {
  type: typeof VIEW_APPRECIATION_CARD;
  displayCardView: boolean;
  cardToView?: AlertModel;
}

export interface AddAppreciationCardsToGroupAction {
  type: typeof LOAD_CARDS_IN_GROUP;
  appreciationCards: AlertModel[];
  cardsType: string;
}
