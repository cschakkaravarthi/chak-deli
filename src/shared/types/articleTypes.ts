import Article from '../models/Article.model';
import Facet, { FacetProperty } from '../models/Facet.model';
import { VariousContentModel } from './contentTypes';

export const LOAD_ARTICLES = 'LOAD_ARTICLES';
export const LOAD_ARTICLE_BY_ID = 'LOAD_ARTICLE_BY_ID';
export const FETCH_MORE_ARTICLES = 'LOAD_MORE_ARTICLES';
export const SET_ARTICLE_FILTER_FACET = 'SET_ARTICLE_FILTER_FACET';
export const CLEAR_ARTICLES_LIST = 'CLEAR_ARTICLES_LIST';
export const LIKE_ARTICLE_TOGGLE = 'LIKE_ARTICLE_TOGGLE';

export interface ArticleState {
  article?: Article;
  isLastPage: boolean;
  articleList: Article[];
  facets: Facet;
  selectedFacet: string;
  pageCount: number;
}

export interface LoadArticlesAction {
  type: typeof LOAD_ARTICLES;
  articleList: Article[];
}

export interface LoadArticleByIdAction {
  type: typeof LOAD_ARTICLE_BY_ID;
  article: VariousContentModel;
}
export interface LoadMoreArticlesAction {
  type: typeof FETCH_MORE_ARTICLES;
  articleList?: Article[];
  facets?: Facet;
}
export interface SetFilterFacetAction {
  type: typeof SET_ARTICLE_FILTER_FACET;
  selectedFacet: FacetProperty;
}

export interface ClearArticlesListAction {
  type: typeof CLEAR_ARTICLES_LIST;
}

export interface LikeArticleToggleAction {
  type: typeof LIKE_ARTICLE_TOGGLE;
  articleId: number;
  likedByUser: boolean;
}
