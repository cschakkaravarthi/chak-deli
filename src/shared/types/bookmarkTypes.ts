import Bookmark from '../models/Bookmark.model';

export const LOAD_BOOKMARKS = 'LOAD_BOOKMARKS';
export const FETCH_BOOKMARKS = 'FETCH_BOOKMARKS';
export const SET_BOOKMARK = 'SET_BOOKMARK';
export const SAVE_BOOKMARK = 'SAVE_BOOKMARK';
export const DELETE_BOOKMARK = 'DELETE_BOOKMARK';
export const REMOVE_BOOKMARK = 'REMOVE_BOOKMARK';

export interface LoadBookmarksAction {
  type: typeof LOAD_BOOKMARKS;
  bookmarks: Bookmark[];
}

export interface SetBookmarkAction {
  type: typeof SET_BOOKMARK;
  success: boolean;
}

export interface DeleteBookmarkAction {
  type: typeof DELETE_BOOKMARK;
  success: boolean;
}

export interface BookmarkState {
  bookmarks: Bookmark[];
  success: boolean;
}
