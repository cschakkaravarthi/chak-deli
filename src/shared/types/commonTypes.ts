import RejectModal from './../models/RejectModal';

export const CURRENT_USER_NAME = 'CURRENT_USER_NAME';
export const TOGGLE_GLOBAL_TOAST = 'TOGGLE_GLOBAL_TOAST';
export const SET_REJECT_MODAL_TEXTAREA = 'SET_REJECT_MODAL_TEXTAREA';
export const SET_REJECT_MODAL_ID_ISSHOW = 'SET_REJECT_MODAL_ID_ISSHOW';

export interface ToggleGlobalToastAction {
  type: typeof TOGGLE_GLOBAL_TOAST;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
}

export type CurrentUser = {
  family_name: string;
  given_name: string;
};

export interface SetCurrentUsername {
  type: typeof CURRENT_USER_NAME;
  currentUsername: CurrentUser;
}

export interface CommonState {
  currentUsername: CurrentUser;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
  idAndIsShow: RejectModal;
  textarea: string;
}

export interface SetRejectModalIdIsShow {
  type: typeof SET_REJECT_MODAL_ID_ISSHOW;
  idAndIsShow: RejectModal;
}

export interface SetRejectModalTextarea {
  type: typeof SET_REJECT_MODAL_TEXTAREA;
  textarea: string;
}
