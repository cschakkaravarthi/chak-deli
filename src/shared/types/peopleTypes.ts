import PeopleModel from '../models/People.model';

export const CLEAR_PEOPLE = 'CLEAR_PEOPLE';
export const FETCH_MORE_PEOPLE = 'FETCH_MORE_PEOPLE';

export interface PeopleState {
  pageCount: number;
  isLastPage: boolean;
  people: PeopleModel[];
}

export interface LoadMorePeopleAction {
  type: typeof FETCH_MORE_PEOPLE;
  people?: PeopleModel[];
}

export interface ClearPeopleAction {
  type: typeof CLEAR_PEOPLE;
}
