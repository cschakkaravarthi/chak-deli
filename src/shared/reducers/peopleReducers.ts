import { Reducer } from 'redux';
import { CLEAR_PEOPLE, FETCH_MORE_PEOPLE, PeopleState } from '../types/peopleTypes';
import { PEOPLE_LIMIT } from '../../constants/constants';

export const initialState = {
  people: [],
  pageCount: 1,
  isLastPage: false
};

const peopleReducers: Reducer<PeopleState, any> = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_MORE_PEOPLE:
      return {
        ...state,
        isLastPage: action.people.length === 0 || action.people.length < PEOPLE_LIMIT,
        people: [...state.people, ...action.people],
        pageCount: state.pageCount + 1
      };
    case CLEAR_PEOPLE:
      return {
        ...state,
        people: [],
        pageCount: 1,
        isLastPage: false
      };
    default:
      return state;
  }
};

export default peopleReducers;
