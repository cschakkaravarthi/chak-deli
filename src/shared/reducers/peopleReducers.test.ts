import peopleReducers, { initialState } from './peopleReducers';
import { FETCH_MORE_PEOPLE } from '../types/peopleTypes';
import { dummyPeople } from '../models/People.model';

describe('peopleReducers', () => {
  it('Handles default case', () => {
    const state = peopleReducers(undefined, {});
    expect(state).toEqual(initialState);
  });

  it('Handles LOAD_PEOPLE action', () => {
    const action = { type: FETCH_MORE_PEOPLE, people: [dummyPeople] };
    const state = peopleReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
