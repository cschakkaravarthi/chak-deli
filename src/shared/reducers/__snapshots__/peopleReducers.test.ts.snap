// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`peopleReducers Handles LOAD_PEOPLE action 1`] = `
Object {
  "isLastPage": true,
  "pageCount": 2,
  "people": Array [
    Object {
      "avatar": Object {
        "photo": "",
      },
      "company": "n/a",
      "costCenter": "GER30/DE050075",
      "country": "Germany",
      "department": "Sales",
      "email": "Bastian.vonPfister@studiocanal.de",
      "employeeNumber": "",
      "employeeType": "3rd Party",
      "firstName": "Bastian",
      "fullName": "Bastian vonPfister",
      "lastName": "vonPfister",
      "location": "",
      "loginId": "Balasua2",
      "managerName": "Joerg Bromberg",
      "mobileNumber": "",
      "phoneNumber": "+49 30 810969 213",
      "photo": Object {
        "photo": "",
      },
      "prefLanguage": "",
      "siteCode": "EXTNL",
      "title": "Manager Business Analysis & Commercial P",
      "voipNumber": "",
    },
  ],
}
`;
